Develop

DenisSangi opened the pull request:
О господи проклятый Гит. 
В общем направляю первую попытку по третьему спринту.
Недостатки:
1) Не смог в параметризацию. Сломал мозг но не понял как передать только цвет заказа в качестве параметризованного теста.
2) Тест КурьерБезПароля падает с ошибкой 500. Не понял почему, аналогичнаый тест КурьерБезЛогина проходит норм.

Commit:
add allure report

Commit:
Первая попытка.
1. Не смог в параметризованные тесты. Я так и не понял как передать в качестве параметров только цвет самоката.
2. Тест КурьерБезПароля падает с ошибкой 500. Не знаю почему аналогичный тест КурьерБезЛогина работает норм.

Commit:
Первая попытка.
1. Не смог в параметризованные тесты. Я так и не понял как передать в качестве параметров только цвет самоката.
2. Тест КурьерБезПароля падает с ошибкой 500. Не знаю почему аналогичный тест КурьерБезЛогина работает норм.

pavel-ulan commented on src/main/java/courier/CourierClient.java:
```diff
@@ -0,0 +1,151 @@
+                .body(creds)
+                .when()
+                .post(ROOT);
+                response.then().log().all()
```
А зачем разбивать на 2 части? Если все делаешь в методе, то и делай сразу. А если хочешь что-то возвращать, то для чего тут часть с проверками?

pavel-ulan commented on src/test/java/CourierTest.java:
```diff
@@ -0,0 +1,100 @@
+        assertTrue(created);
+        CourierCredentials creds = CourierCredentials.from(courier);
+        courierId = courierClient.loginCourier(creds);
+        courierClient.delete(courierId);
```
по хорошему, удаление курьера не должно фигурировать в тесте (если тест упадет на каком-то шаге, то курьер просто не удалится).
удаление надо вынести (можешь предварительно проверять, создавался ли курьер или отсеивать еще как-то).

pavel-ulan commented on src/test/java/CourierTest.java:
```diff
@@ -0,0 +1,100 @@
+    public void createExistedCourierAndCheckResponse() {
+        courierClient.createCourier(courier);
+        CourierCredentials creds = CourierCredentials.from(courier);
+        courierClient.createExistedCourier(creds);
```
тебе бы, по хорошему, из client результаты первой половины метода передавать сюда, а вторую половину, где все проверки, проводить в тестовом методе. а то сейчас твои тесты читаются как набор действий. непонятно по коду, ни что они проверяют, ни для чего они созданы.

pavel-ulan reviewed (changes requested):
Где потерялся pom.xml?
На твои вопросы:
1. передавай парой {цвет, ожидаемыйБуленовскийРезультат}
2. Все норм. В АПИху заложен баг, специально что бы вас запутать. Все норм, баги существуют и если твой тест находит баг, это не значит, что ты сделал что-то неверно, а какраз наоборот (но это работает не всегда. иногда тест может быть просто кривой)

Теперь по коду:
разнеси свои методы из client оставив в клиенте первую часть, а все проверки перенеси в тестовые методы, а то сейчас все выглядит капец как непонятно. Ну и поработай с удалением курьера (сделай обработчик или разнеси тесты по нескольким классам, но вынеси удаление курьера из тестового метода).

Commit:
Add pom.xml

Commit:
Исправления:
1. Методы из CourierClient выносить в тестовый класс не стал, просто убрал там разбивку на 2 части;
2. Для дулаения курьера разнёс тесты по классам Позитивные(с удалением созданного курьера) и Негативные (без создания/удаления);
3. Разбил тестирование заказа по классам Создание заказа(куда передал цвета в виде параметров) Получение тела заказа.

pavel-ulan commented on src/test/java/getOrderBodyTest.java:
```diff
@@ -0,0 +1,22 @@
+    public void getOrderBody(){
+        OrderResponse orderResponse = orderClient.getOrder();
+        assertNotNull(orderResponse);
+        //System.out.println(orderResponse);
```
Вот было же все красиво =)
не забывай удалять неинформативные комментарии. (а по поводу sout - пока делаешь - ок. но в проектах такое не приветствуется, как правило для таких целей есть логгеры).

pavel-ulan reviewed (approved):
Хорошо =)
